### PR TITLE
Recore deprecated SwiftSyntaxParser on top of SwiftParser.

### PR DESCRIPTION
**Pull request by DougGregor**
This drops the dependency on the `_InternalSwiftSyntaxParser` library, and the entire C++ parser/AST/libSyntax stack.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please test

**Comment by DougGregor** [comment]
@swift-ci please test

File: Documentation/Create SwiftSyntax Release.md [hunk 1]
```diff
@@ -2,73 +2,10 @@
 
 ## Create release tag
 1. Check out the tag you want to release in the SwiftSyntax repository
```
Comment (benlangmuir) [inline Documentation/Create SwiftSyntax Release.md:4]: This section only contains a "checkout" step at this point, which doesn't really correspond to the title "Create release tag".

File: Documentation/Create SwiftSyntax Release.md [hunk 2]
```diff
@@ -2,73 +2,10 @@
...
-8. Create a tag and push it to your personal GitHub repository
-9. On your personal GitHub repo create new release, select the tag, name it and upload `_InternalSwiftSyntaxParser.xcframework.zip`
-
 ## Smoke test on macOS
 
 1. Change the binary target in `Package.swift` to point to your personal GitHub repo
-2. Open the SwiftSyntax package with an Xcode version that contains no or an incompatible parser library
-3. Update to Latest Package Versions in Xcode
+2. Update to Latest Package Versions in Xcode
 4. Run tests
```
Comment (benlangmuir) [inline Documentation/Create SwiftSyntax Release.md:9]: 4 -> 3

**Comment by DougGregor** [comment]
This is rebased by https://github.com/apple/swift-syntax/pull/1041

**Pull request closed without merging**